src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml

```diff
@@ -87,9 +87,9 @@
 
         <service id="file_locator" class="Symfony\Component\HttpKernel\Config\FileLocator">
             <argument type="service" id="kernel" />
-            <argument>%kernel.root_dir%/Resources</argument>
+            <argument>%kernel.project_dir%/Resources</argument>
```

Comment:
That looks suspicious: `root_dir` and `project_dir` are not equivalent.

Reply:
What will be the equivalent since this is to me the only thing I can think of to replace root_dir here.

Reply:
I had to look in the code what this does: This is the global directory where resources can be loaded from. `%kernel.project_dir%/Resources` and `%kernel.project_dir%` from below are the fallback directories for any resources. So in SF 3 this means
- `app/Resources` can be used to overwrite resources from Bundles
- `app/` can be used to load any resource if they are not found relative to the current directory the source is loaded from

But that behavior has never officially been unsed in a SF 2/3 symfony-standard or SF 4 flex kernel. They are either loaded relative to the current dir or via glob pattern, not by a global fallback.
Overwriting bundle resources should also not be done like this since forever. See https://symfony.com/doc/current/bundles/override.html
It should be done with compiler passes for services or twig namespace (which uses the /templates/ dir) for templates or translation domains for translations.

If we change this to use project_dir, it means the fallback dir is not `app/Resources` as in SF 3 or `src/Resources` in SF 4, but `Resources/` directly in SF 5. Whatever we put here, it does not make sense and is not documented anywhere.

I suggest to deprecate this global fallback dir in SF 4.4 and remove it in 5. Then this problem is gone.
I already have an idea how to implement the deprecation.

Reply:
Thanks for looking into it @Tobion If you have time to implement the deprecation go ahead, if oyu want me to do it, tell me what you think is the best way to do it!

Reply:
I extracted this issue into #31915. Just merge it like this now and we can deal with this proble separately.

Reply:
I reverted the removal of the `kernel.root_dir` parameter from this PR so we can work on #31915 before.